plugins/modules/keycloak_client_rolescope.py

```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) Ansible project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: keycloak_client_rolescope
+
+short_description: Allows administration of Keycloak client roles scope to restrict the usage of certain roles to a other specific client applications.
+
+version_added: 8.6.0
+
+description:
+    - This module allows you to add or remove Keycloak roles from clients scope via the Keycloak REST API.
+      It requires access to the REST API via OpenID Connect; the user connecting and the client being
+      used must have the requisite access rights. In a default Keycloak installation, admin-cli
+      and an admin user would work, as would a separate client definition with the scope tailored
+      to your needs and a user having the expected roles.
+
+    - Client O(client_id) must have O(full_scope_allowed) set to V(false).
+
+    - Attributes are multi-valued in the Keycloak API. All attributes are lists of individual values and will
+      be returned that way by this module. You may pass single values for attributes when calling the module,
+      and this will be translated into a list suitable for the API.
+
+attributes:
+    check_mode:
+        support: full
+    diff_mode:
+        support: full
+
+options:
+    state:
+        description:
+            - State of the role mapping.
+            - On V(present), all roles in O(role_names) will be mapped if not exists yet.
+            - On V(absent), all roles mapping in O(role_names) will be removed if it exists.
+        default: 'present'
+        type: str
+        choices:
+            - present
+            - absent
+
+    realm:
+        type: str
+        description:
+            - The Keycloak realm under which clients resides.
+        default: 'master'
+
+    client_id:
+        type: str
+        required: true
+        description:
+            - Roles provided in O(role_names) while be added to this client scope.
+
+    client_scope_id:
+        type: str
+        description:
+            - If the O(role_names) are client role, the client id under which it resides.
```

Comment:
```suggestion
            - If the O(role_names) are client role, the client ID under which it resides.
```

Reply:
wow that was fast, thanks

Reply:
I just happened to look a minute after you created this PR ;)